### PR TITLE
EmptyVolume SizeLimit validation and tests

### PR DESCRIPTION
**Pull request by sbezverk**
Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

See for reference issue: #72169

/sig api-machinery
/sig storage
/kind cleanup
/release-note-none

```release-note
None
```

### CODE REVIEW

File: pkg/api/pod/util_test.go [hunk 1]
```diff
@@ -455,17 +456,17 @@ func TestDropRuntimeClass(t *testing.T) {
 		{
 			description:            "pod Without RuntimeClassName",
 			hasPodRuntimeClassName: false,
-			pod:                    podWithoutRuntimeClass,
+			pod: podWithoutRuntimeClass,
```
Comment (liggitt) [inline pkg/api/pod/util_test.go]: check your go version and make sure gofmt is run... verify failed on this

**Comment by liggitt** [comment]
gofmt needs fixing
lgtm otherwise, thanks!

File: pkg/api/pod/util_test.go (hunk 1 above)
Comment (sbezverk) [reply pkg/api/pod/util_test.go]: done

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -412,14 +412,8 @@ func validateVolumeSource(source *core.VolumeSource, fldPath *field.Path, volNam
 	allErrs := field.ErrorList{}
 	if source.EmptyDir != nil {
 		numVolumes++
-		if !utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
```
Comment (liggitt) [inline pkg/apis/core/validation/validation.go]: I just noticed there are four more places `utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation)` is checked in validation and "disabled by feature-gate" errors returned. Those should be dropped.

Comment (sbezverk) [reply pkg/apis/core/validation/validation.go]: Done

**Pull request merged**